src/main/java/com/mongodb/spark/sql/connector/read/partitioner/ShardedPartitioner.java

```diff
@@ -105,6 +107,21 @@ public List<MongoInputPartition> generatePartitions(final ReadConfig readConfig)
       return new SinglePartitionPartitioner().generatePartitions(readConfig);
     }
 
+    BsonDocument keyDocument = configCollectionMetadata.getDocument("key", new BsonDocument());
+    if (keyDocument.keySet().size() > 1) {
+      LOGGER.warn(
+          "Compound shard key: '{}'. Not supported, please use an alternative partitioner.",
+          keyDocument.toJson());
+      throw new MongoSparkException(
+          "Invalid partitioner strategy. The Sharded partitioner does not support compound shard keys.");
+    } else if (keyDocument.containsValue(new BsonString("hashed"))) {
+      LOGGER.warn(
+          "Hashed shard key: '{}'. Not supported, please use an alternative partitioner.",
+          keyDocument.toJson());
+      throw new MongoSparkException(
+          "Invalid partitioner strategy. The Sharded partitioner does not support hashed shard keys.");
```

Comment:
Why do we need to both log and throw here? `FieldPartitioner.createMongoInputPartitions` throws and does not duplicate the error message in logs.

If we indeed need to duplicate messages, should these messages be `ERROR`, given that the connector actually fails?

Reply:
Unlike the other loggings where a single partitioner is used instead, here we error. So we don't need the logs at all.

Removed.